test/e2e/main_test.go

```diff
@@ -61,3 +72,27 @@ func createOperator() *operator.Operator {
 
 	return op
 }
+
+func createTestNamespace() *v1.Namespace {
+	ns := &v1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: e2eTestNamespace,
+		},
+	}
+	if err := f.K8sClient.Create(context.Background(), ns); err != nil {
+		log.Println(err)
+		os.Exit(1)
```

Comment:
imho, it may be useful to just return error if and have a single place to exit - i.e. main so that TestMain reads soething like ... 

```go
func TestMain() {
  if err := main(t); err != nil {
    os.Exit(1)
  }
  os.Exit(0)
}
```